### PR TITLE
Feature/93 adding game over screen

### LINKED ISSUE #93: Implementing a Game Over screen.
**Description**
A Game Over-screen that shows up if you've lost.

**Subtasks**
- [ ] The screen specifies why you lost (i.e. did you go bankrupt? Was the tavern destroyed?)
- [ ] Button that takes you back to Main Menu.
- [ ] Button that let's your restart the game.

### PR DESCRIPTION
**Pull request by DreamingJohnny**
Closes #93 

**Description**  
Set up a simple Game Over screen, I added an enum called "LoseCondition" that checks in what way you lost.

### CODE REVIEW

**Review by MrJackey — CHANGES_REQUESTED** [review]
Is constant pixel size canvas scaling what we want? Also two smaller questions.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 1]
```diff
@@ -71,33 +79,36 @@ public class RoundController : SingletonBehaviour<RoundController> {
...
-
-			foreach (PlayerComponent player in PlayerManager.Instance.Players) {
-				PlayerInput playerInput = player.GetComponent<PlayerInput>();
-				playerInput.SwitchCurrentActionMap("UI");
-			}
 		}
 
-		private void HandleOnNextRound() {
+		private void EnableGamePlay() {
 			VikingController.Instance.CanSpawn = true;
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/RoundController.cs:102]: Your not checking for the instance here, might want to?

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 2]
```diff
@@ -71,33 +79,36 @@ public class RoundController : SingletonBehaviour<RoundController> {
 			VikingController.Instance.StatScaling = new VikingScaling(difficulty, currentRound);
 		}
 
+		private void DisableGamePlay() {
+			VikingController.Instance.CanSpawn = false;
```
Comment (MrJackey) [inline Assets/00_Content/Scripts/Rounds/RoundController.cs:83]: Your not checking for the instance here, might want to?

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 3]
```diff
@@ -111,12 +122,20 @@ public class RoundController : SingletonBehaviour<RoundController> {
...
 
-		private void HandleOnTavernDestroyed() {
-			Debug.Log("Tavern was destroyed!");
+		private void TavernBankrupt() {
+			loseCondition = LoseCondition.Bankrupcy;
+			isRoundActive = false;
+			DisableGamePlay();
+
+			gameOverPanel.Show(loseCondition);
 		}
```
[the remark this answers is not in the document]
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs:139]: Yeah, no reason at all. I've fixed it.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs [hunk 4]
```diff
@@ -111,12 +122,20 @@ public class RoundController : SingletonBehaviour<RoundController> {
...
 
-		private void HandleOnTavernDestroyed() {
-			Debug.Log("Tavern was destroyed!");
+		private void TavernBankrupt() {
+			loseCondition = LoseCondition.Bankrupcy;
+			isRoundActive = false;
+			DisableGamePlay();
```
[the remark this answers is not in the document]
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs:136]: Yeah, good catch, you can only get to here if the input etc. is disabled.

File: Assets/00_Content/Scripts/Rounds/RoundController.cs (hunk 2 above)
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs:83]: Yeah, done!

File: Assets/00_Content/Scripts/Rounds/RoundController.cs (hunk 1 above)
Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Rounds/RoundController.cs:102]: Done

**Review by MrJackey — APPROVED** [review]
:shipit:

**Pull request merged**